### PR TITLE
Clarify that AAD pod identity and workload identity are in preview

### LINKED ISSUE #2952: AAD Pod Identity should be updated with AAD Workload-Identity
**Describe the issue**
Not a bug, doc to be updated

**URL of the docs**
[https://docs.dapr.io/reference/components-reference/supported-secret-stores/azure-keyvault/#configure-the-component](https://docs.dapr.io/reference/components-reference/supported-secret-stores/azure-keyvault/#configure-the-component)

Create and use a managed identity / pod identity by following [this guide](https://docs.microsoft.com/azure/aks/use-azure-ad-pod-identity#create-a-pod-identity). After creating an AKS pod identity, [give this identity read permissions on your desired KeyVault instance](https://docs.microsoft.com/azure/key-vault/general/assign-access-policy?tabs=azure-cli#assign-the-access-policy), and finally in your application deployment inject the pod identity via a label annotation:

**Expected content**
AAD Pod Identity will not go GA, the doc should reflect it and link to AAD WI even if it is in Preview [https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview](https://learn.microsoft.com/en-us/azure/aks/workload-identity-overview)

Also I wonder how to use Dapr Secret store / Azure KV with Azure Container Apps because in ACA you can assign several Identities to your App. 
see also [https://github.com/microsoft/azure-container-apps/issues/7](https://github.com/microsoft/azure-container-apps/issues/7) 

**Screenshots**
N/A

**Additional context**
see also #1547

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

- Refresh whole doc
- Add clarification around pod identity being in preview
- Provide alternative workload identity (also in preview, but on track to go GA eventually)
- Add note clarifying the GA/stable plans for workload identity

## Issue reference

PR will close: #2952

### CODE REVIEW

**Review by msfussell — APPROVED** [review]
LGTM

**Comment by hhunter-ms** [comment]
@ezYakaEagle442 Does this look good to you?

**Comment by hhunter-ms** [comment]
@berndverst per Azure Workload Identity going GA, will my changes here be unnecessary?

**Comment by berndverst** [comment]
@hhunter-ms workload identity in AKS will GA on 3/29. The current version of the workload identity feature in AKS had an issue which didn't support the Dapr sidecar. My fix is making it into their GA on 3/29.
I recently implemented workload identity in Dapr. It will be available / stable in Dapr 1.11

**Comment by hhunter-ms** [comment]
> @hhunter-ms workload identity in AKS will GA on 3/29. The current version of the workload identity feature in AKS had an issue which didn't support the Dapr sidecar. My fix is making it into their GA on 3/29. I recently implemented workload identity in Dapr. It will be available / stable in Dapr 1.11

@berndverst does this PR look ok to you for now then? I did a freshness pass on the whole doc, but the actual new content re workload identity starts line 287

File: daprdocs/content/en/reference/components-reference/supported-secret-stores/azure-keyvault.md [hunk 1]
```diff
@@ -274,12 +279,21 @@ To use **Azure managed identity**:
...
+1. Create and assign a managed identity at the pod-level via either:
+   - [Azure AD workload identity](https://learn.microsoft.com/azure/aks/workload-identity-overview) (preferred method)
+   - [Azure AD pod identity](https://docs.microsoft.com/azure/aks/use-azure-ad-pod-identity#create-a-pod-identity)  
+
+
+   > **Important**: While both Azure AD pod identity and workload identity are in preview, currently Azure AD Workload Identity is planned for general availability (stable state).
+
+1. After creating a workload identity, give it `read` permissions on:
+   - [Your desired KeyVault instance](https://docs.microsoft.com/azure/key-vault/general/assign-access-policy?tabs=azure-cli#assign-the-access-policy)
+   - In your application deployment inject the pod identity via a label annotation.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-secret-stores/azure-keyvault.md]: via a label annotation and by specifying the Kubernetes service account associated with the desired workload identity.

Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-secret-stores/azure-keyvault.md]: Not sure if we want to mention this or not - but unlike when using managed identity directly (where in theory somebody can have multiple identities associated with an app and hence must specify `azureClientId` to specify which identity should be used, with Workload Identity this attribute is not necessary (and has no effect). The Azure identity to be used is inferred from the service account which is tied to an Azure identity via the Azure federated identity.

**Comment by berndverst** [comment]
LGTM - left comments on some more technical details regarding workload identity. Up to you whether you want to include it.

Once Workload Identity is no longer in preview we should update this page. Nobody should use Pod Identity at that time.

**Comment by szuler** [comment]
Please update documentation for Managed Identity via Workload Identity:
for WorkloadIdentity label should be _azure.workload.identity/use: "true"_ and not "_aadpodidbinding: $POD_IDENTITY_NAME_".
See: #[1567](https://github.com/dapr/components-contrib/issues/1567)

**Pull request merged**